### DISCUSSION TITLE
Issue while Upgrading Serverless with @3.2.1

### DISCUSSION THREAD

**Discussion by SaitejaChavva** [serverless/serverless#10733] [Q&A]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

Hi Team,

I was facing issue while upgrading Serverless version with @3.2.1.

**PFB error message which we are getting while Jenkins deployment**
Error:
12:47:32  Cannot resolve serverless.yml: Variables resolution errored with:
12:47:32    - Cannot resolve variable at "functions.get-communication-permission.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.update-communication-permission.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-scheduled-maintenance-ETD.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-scheduled-maintenance-DMY.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-customervehicle-subscriptions.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-vehicle-specific-subscription.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.set-preferred-dealer.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-preferred-dealer.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-vehicle-specifications.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-scheduled-maintenance-vin.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-service-records-self.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.add-service-records-self.events.0.http.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.update-service-records-self.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.remove-service-records-self.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-toyota-care.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-safety-campaigns.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-model-year-list.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-manuals.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-Telematics-Info.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-op-codes.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-customer-vehicles.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.add-customer-vehicles.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.remove-customer-vehicle.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.update-customer-vehicle.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-service-records-summary.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.vehicle-health-report.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.vhr-pdf.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.vehicle-health-history.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.vehicle-mileage-history.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.add-vehicle-driver.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-vehicle-driver.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.delete-vehicle-driver.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.master-vehicles-of-interest.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-vehicles-of-interest.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.update-vehicles-of-interest.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-communication-profile.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.update-communication-profile.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-communication-preferences.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.get-master-preferences.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.update-communication-preferences.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.health-check.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.cloud-watch-log-retention-setter.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.log-export.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.log-cleaner.events.0.http.cors.origins": Value not found at "self" source,
12:47:32    - Cannot resolve variable at "functions.log-subscribe.events.0.http.cors.origins": Value not found at "self" source
12:47:32  make: *** [Makefile.deployServerless:25: deploy] Error 1

Please give me any suggestions and do the needful, how to fix the issue

### Service configuration (serverless.yml) content

```yaml
N/A
```

### Command name and used flags

N/A

### Command output

```shell
N/A
```

### Environment information

```shell
N/A
```

**Comment by pgrzesik** [comment]
Hello, it's very hard to give any suggestions without seeing how your configuration looks like. As it seems like a Q&A topic, I'm going to move this to discussions.

**Reply by SaitejaChavva** [reply]
Hi,

Greetings of the Day!

I cant share all the configurations from my application. Please let me know which part of configuration you need to give any suggestions on the issue.

**Reply by pgrzesik** [reply]
I'm guessing that the problem is with how you configure/access variables that are throwing errors as listed above. Can you share a small reproducible example (not necessarily all your config, but a small working piece with sensitive parts masked out) that has the same problems? It would be easier to diagnose the problem
